### ISSUE TITLE
Keyboard accessibility issues on Android (TextInput but also beyond)

### ISSUE THREAD

**Issue by BogdanCerovac** [facebook/react-native#45801]
### Description

Can not operate TextInput with external keyboards and it seems that TextInputs  "block" focus for other elements if those are between two TextInputs.

We can not really use TextInputs for applications that need to be accessible (app is failing WCAG and EN 301 549 accessibility standards).

This issue is also reopening https://github.com/facebook/react-native/issues/31820 (that was closed because it was stale)...

### Steps to reproduce

1. Connect a physical keyboard to Android device (I've used both PC with direct connection to Android and external Bluetooth keyboard that works with other apps).

2. Open Expo snack (https://snack.expo.dev/aAQHfOdiVoSZF6jo6TzWN) or basically run this code:

```
<Button
  onPress={() => onPressLearnMore("First")}
  title="First focus"
  color="#841584"
/>
      <TextInput
        style={styles.input}
        onChangeText={onChangeText}
        value={text}
      />

      <Button
  onPress={() => onPressLearnMore("between")}
  title="Focus in between"
  color="#2196F3"
/>
      <TextInput
        style={styles.input}
        onChangeText={onChangeNumber}
        value={number}
        placeholder="1234"
        keyboardType="numeric"
      />

            <Button
  onPress={() => onPressLearnMore("Last")}
  title="Last focus"
  color="#000"
/>
```
3. try to focus in either of TextInput's or try to focus and activate the button in between them....

### React Native Version

0.74.3

### Affected Platforms

Runtime - Android

### Output of `npx react-native info`

```text
N/A
```

### Stacktrace or Logs

```text
Nothing reported...
```

### Reproducer

https://snack.expo.dev/aAQHfOdiVoSZF6jo6TzWN

### Screenshots and Videos

https://www.youtube.com/watch?v=EZmllQhzzMU

**Comment by BogdanCerovac** [comment]
As a temporary solution we can use https://www.npmjs.com/package/react-native-external-keyboard or https://github.com/ArturKalach/react-native-a11y from @ArturKalach.

But would be great if React Native would just support this out of the box as many don't test for external keyboard support and therefore introduce barriers to folks without being aware.

**Comment by Sheiicker** [comment]
> I worked on a workaround that allows us to tab between inputs.
> Here is how i made it:
> 
> - Wrap `TextInput` in a homemade `CustomInput` with something like `Pressable` or `TouchableHighlight`. (I used the `TouchableHighlight` from react-native-gesture-handler lib)
> - You can now set your own method `focus` in `CustomInput` which will switch on/off the prop `editable` of your `TextInput` instance. You also should add a reference focusing of your `TextInput` with a callback to focus it.
> 
> Note that when the prop `editable` is set to `false` in your `TextInput`, you can press tab on each TouchableHighlight and go to the next element.
> 
> **Explanation:**
> TouchableHighlight is a focusable element, we can tab through it. The TextInput inside without editable prop will lock the TouchableHighlight and make it impossible to exit. `editable` or `disabled` prop unlocks the navigation. 

 _Originally posted by @Sheiicker in [#31820](https://github.com/facebook/react-native/issues/31820#issuecomment-2665700366)_

**Comment by personalizedrefrigerator** [comment]
[A recent commit](https://github.com/facebook/react-native/commit/3420eb87b01f42092c1691a1f6544f20d244cc60) may address a similar issue — it states
> Changelog: [Android] [Fixed] - TextInputs can now receive focus via external keyboard

**Comment by BogdanCerovac** [comment]
Checked with React Native 0.83.1 and it physical keyboard focus worked well on TextInput. Discovered another issue, will open a separate one...

Demo video: https://youtu.be/mrW2aJZ5SLc

Test code:

```
import {
  StatusBar,
  StyleSheet,
  useColorScheme,
  View,
  Button,
  TextInput,
  Text,
} from 'react-native';
import {
  SafeAreaProvider,
  useSafeAreaInsets,
} from 'react-native-safe-area-context';
import React, { useState } from 'react';

function App() {
  const isDarkMode = useColorScheme() === 'dark';

  return (
    <SafeAreaProvider>
      <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
      <AppContent />
    </SafeAreaProvider>
  );
}

function AppContent() {
  const [text, onChangeText] = useState('');
  const [number, onChangeNumber] = useState(null);

  const [focusedInput, setFocusedInput] = useState(null);

  const onPressLearnMore = (which) => {
    alert('useless button, indeed. ' + which + ' pressed');
  };

  const safeAreaInsets = useSafeAreaInsets();

  return (
    <View style={styles.container} style={{ padding: 10 }}>
      <Text style={styles.h1} accessibilityRole="header">
        Testing TextInput for accessibility
      </Text>

      <Text style={{ margin: 10 }}>30th of Dec. 2025</Text>

      <Text style={{ margin: 10 }}>
        Currently focused input field: {focusedInput}
      </Text>

      <Button
        onPress={() => onPressLearnMore('First')}
        onBlur={() => setFocusedInput(null)}
        title="First focus"
        color="#841584"
      />
      <TextInput
        accessibilityLabel="Your GUID (random text)"
        style={styles.input}
        onChangeText={onChangeText}
        onFocus={() => setFocusedInput('first input')}
        onBlur={() => setFocusedInput(null)}
        value={text}
      />

      <Button
        onPress={() => onPressLearnMore('between')}
        onBlur={() => setFocusedInput(null)}
        title="Focus in between"
        color="#2196F3"
      />
      <TextInput
        accessibilityLabel="Your personal code (4 digits)"
        style={styles.input}
        onChangeText={onChangeNumber}
        onFocus={() => setFocusedInput('second input')}
        onBlur={() => setFocusedInput(null)}
        value={number}
        keyboardType="numeric"
      />

      <Button
        onPress={() => onPressLearnMore('Last')}
        onBlur={() => setFocusedInput(null)}
        title="Last focus"
        color="#000"
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  input: {
    height: 40,
    margin: 12,
    borderWidth: 1,
  },
  h1: {
    fontWeight: 'bold',
    fontSize: 30,
    marginTop: 40,
  },
});

export default App;

```
